Frontend/src/app/graph-panel/page.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+import React, { useState } from "react";
+import { InboxOutlined } from "@ant-design/icons";
+import type { UploadProps } from "antd";
+import {
+  Button,
+  Col,
+  Divider,
+  message,
+  Row,
+  Typography,
+  Upload,
+  Modal,
+  Input,
+} from "antd";
+import kafkaLOGO from "@/assets/images/kafka-logo.jpg";
+import hadoopLOGO from "@/assets/images/hadoop-logo.jpg";
+import Image from "next/image";
+import KafkaUploadModal from "@/components/graph-panel/kafka-upload-modal";
+import HadoopUploadModal from "@/components/graph-panel/hadoop-upload-modal";
+import { RcFile } from "antd/es/upload/interface";
+import { toast } from "react-toastify";
+import axios from "axios";
+
+const { Dragger } = Upload;
+
+export default function GraphUpload() {
+  const [kafkaModalOpen, setKafkaModelOpen] = useState<boolean>(false);
+  const [hadoopModalOpen, setHadoopModelOpen] = useState<boolean>(false);
+  const [file, setFile] = useState<File>();
+  const [fileUrl, setFileUrl] = useState<string>();
+  const [modalOpen, setModalOpen] = useState(false);
+  const [graphName, setGraphName] = useState<string>("");
+
+  const handleFileUpload = (file: RcFile) => {
+    setModalOpen(true);
+    setFile(file);
+
+    if (fileUrl) {
+      URL.revokeObjectURL(fileUrl);
+    }
+
+    if (file) {
+      const url = URL.createObjectURL(file);
+      setFileUrl(url);
+    } else {
+      setFileUrl(undefined);
+    }
+    return false;
+  };
+
+  const handleUpload = async () => {
+    if (!file) {
+      toast.error("Please select a file to upload");
+      return;
+    }
+
+      const formData = new FormData();
+      formData.append('file', file);    
+      formData.append('graphName', graphName); // Append the file name
+
+      // Send the file and filename with Axios POST request
+      axios.post('/backend/graph/upload', formData, {
```

Comment:
Is there a file size limit for the files getting uploaded? Although JasmineGraph frontend does not impose such limit, we need to check whether the axios library imposes such limit.

Reply:
Axios itself doesn’t impose a file size limit, so any upload limits would come from the backend server configuration or available system memory.

Reply:
Ok